### PR TITLE
Fix ILRepack not execute on netcoreapp

### PR DESCRIPTION
**Pull request by pengweiqhca**
```
System.IO.FileNotFoundException: Could not load file or assembly 'Mono.Cecil, Version=0.11.1.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e'.
```

### CODE REVIEW

**Comment by pardeike** [comment]
You change fails in AppVeyor CI:

```
Build FAILED.
2350
2351"C:\projects\harmony\Harmony.sln" (default target) (1) ->
2352"C:\projects\harmony\Harmony\Harmony.csproj" (default target) (2) ->
2353(ILRepack target) -> 
2354  C:\projects\harmony\Harmony\Harmony.csproj(104,3): error : Failed to resolve System.Reflection.BindingFlags
2355C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.Mixin.CheckedResolve(TypeReference self)
2356C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.GetConstantType(TypeReference constant_type, Object constant)
2357C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddConstant(IConstantProvider owner, TypeReference type)
2358C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddField(FieldDefinition field)
2359C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddFields(TypeDefinition type)
2360C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type)
2361C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddTypes()
2362C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildTypes()
2363C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildModule()
2364C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildMetadata()
2365C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.BuildMetadata(ModuleDefinition module, MetadataBuilder metadata)
2366C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
2367C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
2368C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters)
2369C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.AssemblyDefinition.Write(String fileName, WriterParameters parameters)
2370C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at ILRepacking.ILRepack.Repack()
2371C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at ILRepack.MSBuild.Task.ILRepack.Execute()
```

**Comment by pengweiqhca** [comment]
> You change fails in AppVeyor CI:
> 
> ```
> Build FAILED.
> 2350
> 2351"C:\projects\harmony\Harmony.sln" (default target) (1) ->
> 2352"C:\projects\harmony\Harmony\Harmony.csproj" (default target) (2) ->
> 2353(ILRepack target) -> 
> 2354  C:\projects\harmony\Harmony\Harmony.csproj(104,3): error : Failed to resolve System.Reflection.BindingFlags
> 2355C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.Mixin.CheckedResolve(TypeReference self)
> 2356C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.GetConstantType(TypeReference constant_type, Object constant)
> 2357C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddConstant(IConstantProvider owner, TypeReference type)
> 2358C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddField(FieldDefinition field)
> 2359C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddFields(TypeDefinition type)
> 2360C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddType(TypeDefinition type)
> 2361C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.AddTypes()
> 2362C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildTypes()
> 2363C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildModule()
> 2364C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.MetadataBuilder.BuildMetadata()
> 2365C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.BuildMetadata(ModuleDefinition module, MetadataBuilder metadata)
> 2366C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.Write(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
> 2367C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleWriter.WriteModule(ModuleDefinition module, Disposable`1 stream, WriterParameters parameters)
> 2368C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters)
> 2369C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at Mono.Cecil.AssemblyDefinition.Write(String fileName, WriterParameters parameters)
> 2370C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at ILRepacking.ILRepack.Repack()
> 2371C:\projects\harmony\Harmony\Harmony.csproj(104,3): error :    at ILRepack.MSBuild.Task.ILRepack.Execute()
> ```
I can't reappear in my computer. It is ilpack error not test.

File: Harmony/Harmony.csproj [hunk 1]
```diff
@@ -77,7 +64,7 @@
 		<PackageReference Include="jnm2.ReferenceAssemblies.net35" Version="1.0.1" PrivateAssets="all" Condition="'$(TargetFramework)'=='net35'">
 			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
 		</PackageReference>
-		<PackageReference Include="ILRepack.MSBuild.Task" Version="2.0.13" PrivateAssets="All" />
+        <PackageReference Include="SourceLink.Copy.PdbFiles" Version="2.8.3" PrivateAssets="All" />
```
Comment (0x0ade) [inline Harmony/Harmony.csproj]: What is the reason for replacing `ILRepack.MSBuild.Task` with `SourceLink.Copy.PdbFiles`? From what I can see they both solve very different issues...

Comment (pengweiqhca) [reply Harmony/Harmony.csproj]: `ILRepack.MSBuild.Task` is repeated

**Comment by pengweiqhca** [comment]
```
System.IO.FileNotFoundException: Could not load file or assembly 'Mono.Cecil, Version=0.11.1.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e'.
File name: 'Mono.Cecil, Version=0.11.1.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e'
   at HarmonyLib.HarmonySharedState.CreateModule()
   at HarmonyLib.HarmonySharedState.GetState()
   at HarmonyLib.HarmonySharedState.GetPatchInfo(MethodBase method)
   at HarmonyLib.PatchProcessor.Patch()
   at HarmonyLib.Harmony.Patch(MethodBase original, HarmonyMethod prefix, HarmonyMethod postfix, HarmonyMethod transpiler, HarmonyMethod finalizer)
```
image

**Comment by pengweiqhca** [comment]
https://github.com/pardeike/Harmony/issues/255

**Pull request closed without merging**